activerecord/test/cases/base_test.rb

```diff
@@ -847,9 +847,8 @@ def test_dup_with_aggregate_of_same_name_as_attribute
     assert_not_predicate dup, :persisted?
 
     # test if the attributes have been duped
-    original_amount = dup.salary.amount
-    dev.salary.amount = 1
-    assert_equal original_amount, dup.salary.amount
+    dup.salary = DeveloperSalary.new(2 * dup[:salary])
+    assert_not dev.salary.equal?(dup.salary)
```

Comment:
I'd prefer if you didn't change the test.

Reply:
I had to because `dev.salary` is frozen (that's the point of the PR), so `dev.salary.amount = 1` is going to raise.

Or am I missing something?

Reply:
Right sorry. But ideally you'd just invert it:
```ruby
    # test if the attributes have been duped
    original_amount = dup.salary.amount
    dup.salary.amount = 1
    assert_equal original_amount, dev.salary.amount
```

Reply:
Hum, sorry I'm tired,...I don't think that cover the code as well.

What I'd like to see is:

```ruby
    salary = DeveloperSalary.new(42)
    dup.salary = salary
    salary.amount = 1
    assert_equal 42, dup.salary
```

What I want to show is that `salary=` didn't freeze the argument, and just kept a copy of it.